geo/src/algorithm/is_convex.rs

```diff
@@ -189,13 +189,12 @@ where
 
     // Now we have a fixed orientation expected at the rest
     // of the coords. Loop to check everything matches it.
-    if ((i + 1)..n)
+    if !((i + 1)..n)
```

Comment:
To me, this still seems hard to read :sweat_smile:.

Reply:
Does:

```rust
if !((i + 1)..n)
 
 lnicola 43 seconds ago Member

To me, this still seems hard to read.

Does something like

```rust
    if ((i + 1)..n)
        .map(orientation_at)
        .all(|(_, orientation)| match orientation {
            Orientation::Collinear => allow_collinear,
            orientation => orientation == first_non_collinear,
        })
```

work?

Reply:
Sure, readability can be a bit subjective

Reply:
> Does something like
> 
> ```rust
>     if ((i + 1)..n)
>         .map(orientation_at)
>         .all(|(_, orientation)| match orientation {
>             Orientation::Collinear => allow_collinear,
>             orientation => orientation == first_non_collinear,
>         })
> ```
> 
> work?

I didn't want to touch the match, but sure, it's equivalent

Reply:
I'm not actually sure, but hopefully it's covered by tests? I think it's worth applying DeMorgan's in this case.

Reply:
Of course, it is a simplification

Reply:
As phrased in the current PR, this isn't an obvious improvement to me, and every time we touch code people have to spend time reviewing it, and of course we might break something that was working before, so I'd caution you against work that changes presumed working code for stylistic reasons unless there is a very clear and sizable readability vs. risk payoff.

(I think the other two changes meet this criteria FWIW)

I'd actually prefer you just revert this part altogether, but if other folks think this code should change, I think @lnicola's suggestion is the best so far, provided you're willing to verify the test cases cover all possible paths relevant to the change.
```
if ((i + 1)..n)
        .map(orientation_at)
        .all(|(_, orientation)| match orientation {
            Orientation::Collinear => allow_collinear,
            orientation => orientation == first_non_collinear,
        })
```

Reply:
I could guarantee that this change is equivalent, but I think that is not the point. I fully agree not to change the code that works without objective reasons. Thanks everyone for the feedback, I'm going to revert this part.

Reply:
Let me know anything else.